File: Convert Sorted Array to Binary Search Tree/Convert_Sorted_Array_to_Binary_Search_Tree.py
# 116ms 19.37%

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        def helper(left, right):
            if left > right:
                return None
            else:
                mid = (left + right) // 2

                root = TreeNode(nums[mid])
                root.left = helper(left, mid - 1)
                root.right = helper(mid + 1, right)

                return root

        return helper(0, len(nums) - 1)

    
        

File: Convert Sorted Array to Binary Search Tree/Convert_Sorted_Array_to_Binary_Search_Tree.cpp
// 12ms 66.86%

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

#include<vector>
using namespace std;

class Solution {
public:
    TreeNode* sortedArrayToBST(vector<int>& nums) 
    {
        return helper(0, nums.size() - 1, nums);
    }

    TreeNode* helper(int left, int right, vector<int>& nums)
    {
        if (left > right)
            return 0;
        else 
        {
            int mid = (left + right) / 2;

            TreeNode *root = new TreeNode(nums[mid]);
            root->left = helper(left, mid - 1, nums);
            root->right = helper(mid + 1, right, nums);

            return root;
        }
    }
};